src/CarbonAware.WebApi/test/unitTests/CarbonAwareControllerTests.cs

```diff
@@ -34,55 +58,54 @@ public async Task GetEmissions_SuccessfulCallReturnsOk()
         };
         var controller = new CarbonAwareController(this.MockCarbonAwareLogger.Object, CreateAggregatorWithEmissionsData(data).Object);
 
-        IActionResult ar1 = await controller.GetEmissionsDataForLocationByTime(location);
-        IActionResult ar2 = await controller.GetEmissionsDataForLocationsByTime(new string[] { location });
+        IActionResult result = await controller.GetEmissionsDataForLocationByTime(location);
 
-        TestHelpers.AssertStatusCode(ar1, HttpStatusCode.OK);
-        TestHelpers.AssertStatusCode(ar2, HttpStatusCode.OK);
+        TestHelpers.AssertStatusCode(result, HttpStatusCode.OK);
     }
 
     /// <summary>
     /// Tests that successful best emissions call to an aggregator with any data returned results in action with OK status.
     /// </summary>
-    [Test]
-    public async Task GetBestEmissions_SuccessfulCallReturnsOk()
+    [TestCase(new object?[] { null, "Sydney" }, TestName = "GetBestEmissions simulates 'location=&location=Sydney'")]
+    [TestCase(new object?[] { "Sydney", null }, TestName = "GetBestEmissions simulates 'location=Sydney&location='")]
+    [TestCase(new object?[] { "Sydney" }, TestName = "GetBestEmissions simulates 'location=Sydney'")]
+    public async Task GetBestEmissions_SuccessfulCallReturnsOk(params string[] locations)
     {
-        string location = "Sydney";
         var data = new EmissionsData()
         {
-            Location = location,
+            Location = "Sydney",
             Rating = 0.9,
             Time = DateTime.Now
         };
 
-
         var controller = new CarbonAwareController(this.MockCarbonAwareLogger.Object, CreateAggregatorWithBestEmissionsData(data).Object);
 
-        var ar = await controller.GetBestEmissionsDataForLocationsByTime(new string[] { location });
+        var result = await controller.GetBestEmissionsDataForLocationsByTime(locations);
 
-        TestHelpers.AssertStatusCode(ar, HttpStatusCode.OK);
+        TestHelpers.AssertStatusCode(result, HttpStatusCode.OK);
     }
 
     /// <summary>
     /// Tests that successful forecast call to an aggregator with any data returned results in action with OK status.
     /// </summary>
-    [Test]
-    public async Task GetForecast_SuccessfulCallReturnsOk()
+    [TestCase(new object?[] { null, "Sydney" }, TestName = "GetForecast simulates 'location=&location=Sydney'")]
```

Comment:
I'm confused here. Why null succeed?

Reply:
We filter out any null values before processing, so by the time we are constructing the `props` dictionary this mixed array has already been reduced to just `[ Location ]`